File: my-react-app/src/models/ProductDto.js
import UserInfoDto from "./UserInfoDto";

class ProductDto {
    constructor(id, name, price, quantity, picture, description, userId, seller, orderedQuantity) {
      this.id = id || 0;
      this.name = name || "";
      this.price = price || 0.0;
      this.quantity = quantity || 0;
      this.picture = picture || null;
      this.description = description || "";
      this.userId = userId || "";
      this.seller = seller===null ? null : new UserInfoDto(seller.username, seller.email, seller.name, seller.surname, seller.address, seller.picture);
      this.orderedQuantity = orderedQuantity || 0;
    }
  }
  
  
export default ProductDto;
  

File: microservices-server-app/ProductOrderWebApi/Dto/ProductDto.cs
using Microsoft.AspNetCore.Http;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace ProductOrderWebApi.Dto
{
    public class ProductDto
    {
        public long Id { get; set; }
        public string Name { get; set; }
        public float Price { get; set; }
        public int Quantity { get; set; }
        public string Description { get; set; }
        public IFormFile PictureFromForm { get; set; }
        public long UserId { get; set; }
    }
}
